lib/client/db/database_certificates.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package db
+
+import (
+	"context"
+	"crypto/x509/pkix"
+	"time"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/client/identityfile"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/trace"
+)
+
+// GenerateDatabaseCertificatesRequest contains the required fields used to generate database certificates
+// Those certificates will be used by databases to set up mTLS authentication against Teleport
+type GenerateDatabaseCertificatesRequest struct {
+	ClusterAPI         auth.ClientI
+	Principals         []string
+	OutputFormat       identityfile.Format
+	OutputCanOverwrite bool
+	OutputLocation     string
+	IdentityFileWriter identityfile.ConfigWriter
+	TTL                time.Duration
+	Key                *client.Key
+}
+
+// GenerateDatabaseCertificates to be used by databases to set up mTLS authentication
+func GenerateDatabaseCertificates(ctx context.Context, req GenerateDatabaseCertificatesRequest) ([]string, error) {
+	if req.OutputFormat != identityfile.FormatSnowflake && len(req.Principals) == 1 && req.Principals[0] == "" {
+		return nil, trace.BadParameter("at least one hostname must be specified")
+	}
+
+	// For CockroachDB node certificates, CommonName must be "node":
+	//
+	// https://www.cockroachlabs.com/docs/v21.1/cockroach-cert#node-key-and-certificates
+	if req.OutputFormat == identityfile.FormatCockroach {
+		req.Principals = append([]string{"node"}, req.Principals...)
+	}
+
+	subject := pkix.Name{CommonName: req.Principals[0]}
```

Comment:
Just noticed that we don't validate that `req.Principals` is not empty anywhere here, might be worth adding a check somewhere here to avoid panics.